lib/ansiblelint/testing/__init__.py

```diff
@@ -31,11 +31,10 @@ def _call_runner(self, path) -> List["MatchError"]:
 
     def run_playbook(self, playbook_text):
         """Lints received text as a playbook."""
-        play_root = tempfile.mkdtemp()
-        with open(os.path.join(play_root, 'playbook.yml'), 'w') as fp:
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".yml", prefix="playbook") as fp:
             fp.write(playbook_text)
-        results = self._call_runner(fp.name)
-        shutil.rmtree(play_root)
+            fp.flush()
```

Comment:
this may be useful too
```suggestion
            fp.flush()  # guarrantee it's on disk before linting
```